src/iter/repeat.rs

```diff
@@ -0,0 +1,108 @@
+use super::*;
+use super::internal::*;
+use std::usize;
+
+pub struct Repeat<T> {
+    element: T,
+}
+
+pub fn repeat<T: Clone>(elt: T) -> Repeat<T> {
+    Repeat{element: elt}
+}
+
+impl<T: Clone> Clone for Repeat<T>{
+    fn clone(&self) -> Repeat<T> {
+        Repeat{ element: self.element.clone()}
+    }
+}
+
+impl<T> ParallelIterator for Repeat<T>
+    where T: Clone + Send
+{
+    type Item = T;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+       where C: UnindexedConsumer<Self::Item>
+   {
+       bridge(self, consumer)
+   }
+}
+
+impl<T> IndexedParallelIterator for Repeat<T>
+    where T: Clone + Send
+{
+    fn drive<C>(self, consumer: C) -> C::Result
+        where C: Consumer<Self::Item>
+    {
+            bridge(self, consumer)
+    }
+
+    fn with_producer<CB>(self, callback: CB) -> CB :: Output
+        where CB: ProducerCallback<Self::Item>
+        {
+            callback.callback(RepeatProducer{ repeat: self})
+        }
+
+    fn len(&mut self) -> usize { usize::MAX }
+}
+
+
+/// Producer
+
+struct RepeatProducer<P> {
+    repeat: Repeat<P>,
```

Comment:
I wonder, why does this store the whole `Repeat` type, and not just the element?  It's a minor difference, but then we don't need to worry about `Clone` for `Repeat` itself.